x-pack/legacy/plugins/epm/common/types.ts

```diff
@@ -61,45 +69,27 @@ export interface AssetParts {
   type: AssetType;
   file: string;
 }
-export type AssetTypeToParts = Record<AssetType, AssetParts[]>;
-export type AssetsGroupedByServiceByType = Record<ServiceName, AssetTypeToParts>;
-export interface RegistryPackage {
-  name: string;
-  version: string;
-  description: string;
-  readme?: string;
-  icon: string;
-  requirement: RequirementsByServiceName;
-  title?: string;
-  screenshots?: ScreenshotItem[];
-  assets: string[];
-}
-
-// Managers public HTTP response types
-export type PackageList = PackageListItem[];
-// add title here until it's a part of registry response
-export type PackageListItem = Installable<Required<RegistryListItem>>;
-export type PackagesGroupedByStatus = Record<InstallationStatus, PackageList>;
-
-// add title here until it's a part of registry response
-export type PackageInfo = Installable<
-  Required<RegistryPackage> & { assets: AssetsGroupedByServiceByType }
->;
-
-export type Installation = SavedObject<InstallationAttributes>;
-export interface InstallationAttributes extends SavedObjectAttributes {
-  installed: AssetReference[];
-}
-
-export type Installable<T> = Installed<T> | NotInstalled<T>;
 
-export type Installed<T = {}> = T & {
-  status: 'installed';
-  savedObject: Installation;
+export type KibanaAssetParts = AssetParts & {
+  service: Extract<'kibana', ServiceName>;
+  type: KibanaAssetType;
 };
 
-export type NotInstalled<T = {}> = T & {
-  status: 'not_installed';
+export type ElasticsearchAssetParts = AssetParts & {
```

Comment:
Not strictly required, but it narrows the types so `service: 'elasticsearch'` can only have `type: ElasticsearchAssetType `